### DISCUSSION TITLE
Some change in CasePath behaviour noticed

### DISCUSSION THREAD

**Discussion by rehsals** [pointfreeco/swift-composable-architecture#666] [Show and tell]
I've been using TCA version `0.19.0` and today I've updated to `0.22.0`.
And instantly I've got several breakpoints about nil-states when corresponding reducer gets an action.

My setup was as follows:
```
struct LocalState {}
enum LocalAction {
  case whatever
}

let localReducer: Reducer<LocalState, LocalAction, Void> = <...>

struct GlobalState {
  enum Option {
    case local(LocalState)
  }

  var option: Option? <--- optionality is important I guess
}

enum GlobalAction {
  case local(LocalAction)
}

let globalReducer = Reducer<GlobalState, GlobalAction, Void>.combine(
  localReducer
    .pullback(
            state: /GlobalState.Option.local,
            action: .`self`,
            environment: { $0 })
    .pullback(
            state: \.option,
            action: /GlobalAction.local,
            environment: { $0 }),
  .init { state, action, _ in <...> }
)
```
So `CasePath` from `GlobalState.Option?` to `LocalState` worked fine on `0.19.0` and not working anymore.
I've changed my code to be:
```
let globalReducer = Reducer<GlobalState, GlobalAction, Void>.combine(
  localReducer
    .pullback(
            state: /GlobalState.Option.local,
            action: .`self`,
            environment: { $0 })
    .pullback(
            state: /Optional<GlobalState.Option>.some,
            action: .`self`,
            environment: { $0 })
    .pullback(
            state: \.option,
            action: /GlobalAction.local,
            environment: { $0 }),
  .init { state, action, _ in <...> }
)
```
and things got back to normal.
Is this is an expected behaviour and a bug being fixed? May be it could be useful for someone in similar situations

**Edit:** probably it's a `CasePath` version what is more relevant. My current is `0.5.0`. Unfortunately I can't tell which version I was using before the update
**Edit2:** there is an issue in `CasePath` repo which seems related: https://github.com/pointfreeco/swift-case-paths/issues/40

**Comment by stephencelis** [comment]
@rehsals This is indeed https://github.com/pointfreeco/swift-case-paths/issues/40. Swift's optional promotion can promote embed functions to optional-returning embed functions, which breaks some of the expectations made in the new runtime code. We hope to workaround this problem when we have time, but in the meantime using an explicit `/Optional.some` is definitely the way to go.

One thing you can do, though, is avoid an extra `pullback` by using `CasePath.appending(path:)` instead:

```swift
    .pullback(
            state: (/Optional.some).appending(path: /GlobalState.Option.local),
            action: .`self`,
            environment: { $0 })
```

**Reply by rehsals** [reply]
Got it. Nice hint for avoiding extra `pullback`, thanks!
